lib/rides/pages/search_ride_page.dart

```diff
@@ -125,133 +105,270 @@ class _SearchRideFormState extends State<SearchRideForm> {
-          value: _dropdownValue,
-          icon: const Icon(Icons.arrow_downward),
-          decoration: InputDecoration(
-            border: const OutlineInputBorder(),
-            labelText: S.of(context).seats,
+        width: 150,
+        child: IncrementField(
+          maxValue: 8,
+          icon: Icon(
+            Icons.chair,
+            color: Theme.of(context).colorScheme.primary,
           ),
           onChanged: (int? value) {
             setState(() {
-              _dropdownValue = value!;
+              _seats = value!;
+              loadRides();
             });
           },
-          items: list.map<DropdownMenuItem<int>>((int value) {
-            return DropdownMenuItem<int>(
-              value: value,
-              child: Text(value.toString()),
-            );
-          }).toList(),
         ),
       ),
     );
   }
 
   @override
   Widget build(BuildContext context) {
-    Widget submitButton = Button.submit(
-      S.of(context).pageSearchRideButtonSearch,
-      onPressed: _onSubmit,
-    );
-    return Form(
-      key: _formKey,
-      child: Padding(
-        padding: const EdgeInsets.fromLTRB(10, 20, 10, 0),
-        child: Column(
-          children: <Widget>[
-            AddressSearchField.start(
-              controller: _startController,
-              onSelected: (AddressSuggestion suggestion) {
-                _startSuggestion = suggestion;
-              },
-            ),
-            const SizedBox(height: 15),
-            AddressSearchField.destination(
-              controller: _destinationController,
-              onSelected: (AddressSuggestion suggestion) {
-                _destinationSuggestion = suggestion;
-              },
-            ),
-            Container(
-              padding: const EdgeInsets.symmetric(vertical: 15),
-              child: Row(
-                mainAxisAlignment: MainAxisAlignment.spaceBetween,
-                children: <Widget>[
-                  buildDatePicker(),
-                  buildTimePicker(),
-                  const SizedBox(width: 50),
-                  buildSeatsPicker(),
+    return Hero(
+      tag: "RideFAB",
+      child: Scaffold(
+        body: SafeArea(
+          child: Padding(
+            padding: const EdgeInsets.all(8.0),
+            child: RefreshIndicator(
+              onRefresh: loadRides,
+              child: CustomScrollView(
+                slivers: <Widget>[
+                  SliverAppBar(
+                    floating: true,
+                    flexibleSpace: FlexibleSpaceBar(
+                      background: Container(color: Theme.of(context).colorScheme.surface),
+                      title: Text(
+                        S.of(context).pageSearchRideTitle,
+                        style: TextStyle(color: Theme.of(context).colorScheme.onSurface),
+                      ),
+                    ),
+                  ),
+                  SliverPinnedHeader(
```

Comment:
Could we also pin the Appbar, just feel like you should always be able to click on back from any position

Reply:
That would be very easy, but it's also very common for Android apps to do this. You can still go back via the gesture or access the app bar immediately if you start to scroll up again